### PR TITLE
Assigned facility: My facilities

### PR DESCRIPTION
**Pull request by prabhanshuguptagit**
**Story card:** https://app.clubhouse.io/simpledotorg/story/480

## Because
We are making changes to our reports to incorporate assigned facility.

## This addresses

This changes the "BP control" and "Missed visits" reports in My facilities to be Assigned facility compliant.

The query changes in detail:
- [BP control](https://docs.google.com/document/d/1YZqm0t5NhTDUkn3Ww8yRqAszKCtAiOqChpFED6U-QXE/edit#heading=h.2b4bqhbe3tf7)
- [Missed visit](https://docs.google.com/document/d/1YZqm0t5NhTDUkn3Ww8yRqAszKCtAiOqChpFED6U-QXE/edit#heading=h.x63c1ja44de)

- BP control footnote before:
image

- BP control footnote after:
image

### CODE REVIEW

File: app/views/my_facilities/blood_pressure_control.html.erb [hunk 1]
```diff
@@ -152,9 +152,8 @@
       <ol>
         <li><strong>Overall BP control:</strong>
           <ul>
-            <li>Numerator: Patients registered at this facility whose most recent BP in the past 90 days was &lt;140/90.</li>
-            <li>Denominator: Patients registered at this facility.</li>
-            <li>Patients registered in the last 2 months are excluded from both the <i>Numerator</i> and <i>Denominator</i>.</li>
+            <li>Numerator: Patients whose last BP in the past 90 days was &lt;140/90.</li>
+            <li>Denominator: Patients with this preferred facility.</li>
```
Comment (rsanheim) [inline app/views/my_facilities/blood_pressure_control.html.erb:156]: Can we use the actual term we are refactoring towards here? "Patients with this assigned facility" ?  Thats what the queries are changing to use, right?

Having multiple terms for the same concept adds confusing for us and for our users when we do training or help docs and the like.

File: app/views/my_facilities/missed_visits.html.erb [hunk 2]
```diff
@@ -141,6 +141,7 @@
           their
           personal phone number or used another phone.
         </li>
+        <li><strong>Total patients:</strong>Patients with this preferred facility.</li>
```
Comment (rsanheim) [inline app/views/my_facilities/missed_visits.html.erb:144]: same question as above

File: app/queries/my_facilities/blood_pressure_control_query.rb [hunk 3]
```diff
@@ -22,6 +20,23 @@ def initialize(facilities: Facility.all, cohort_period: {})
     @facilities = facilities
   end
 
+  def cohort_registrations_per_facility
```
Comment (vkrmis) [inline app/queries/my_facilities/blood_pressure_control_query.rb:23]: Can we rename this? This isn't the count of registrations anymore, so calling it `registrations_per_facility` is confusing.

File: app/queries/my_facilities/blood_pressure_control_query.rb [hunk 4]
```diff
@@ -100,7 +122,7 @@ def quarterly_uncontrolled_bps
   def monthly_registrations
```
Comment (vkrmis) [inline app/queries/my_facilities/blood_pressure_control_query.rb:122]: Same here, this is not counting registrations anymore

File: app/queries/my_facilities/blood_pressure_control_query.rb [hunk 5]
```diff
@@ -73,7 +95,7 @@ def overall_controlled_bps
   def quarterly_registrations
```
Comment (vkrmis) [inline app/queries/my_facilities/blood_pressure_control_query.rb:95]: Can we call this `quarterly_patients`?

File: app/controllers/my_facilities_controller.rb [hunk 6]
```diff
@@ -45,12 +45,12 @@ def blood_pressure_control
                overall_patients: bp_query.overall_patients.count,
                overall_controlled_bps: bp_query.overall_controlled_bps.count}
 
-    @registered_patients_per_facility = bp_query.cohort_registrations.group(:registration_facility_id).count
-    @controlled_bps_per_facility = bp_query.cohort_controlled_bps.group(:registration_facility_id).count
-    @uncontrolled_bps_per_facility = bp_query.cohort_uncontrolled_bps.group(:registration_facility_id).count
+    @registered_patients_per_facility = bp_query.cohort_registrations_per_facility
+    @controlled_bps_per_facility = bp_query.cohort_controlled_bps_per_facility
```
Comment (vkrmis) [inline app/controllers/my_facilities_controller.rb:49]: This looks nice, should we also refactor the overview query and registrations query to be consistent?

File: app/queries/my_facilities/blood_pressure_control_query.rb (hunk 5 above)
Comment (prabhanshuguptagit) [reply app/queries/my_facilities/blood_pressure_control_query.rb:95]: Same as `monthly_registrations`.

File: app/queries/my_facilities/blood_pressure_control_query.rb (hunk 4 above)
Comment (prabhanshuguptagit) [reply app/queries/my_facilities/blood_pressure_control_query.rb:122]: This is actually the total registrations in the month at the facilities, so `monthly_registrations` makes sense.

Comment (vkrmis) [reply app/queries/my_facilities/blood_pressure_control_query.rb:122]: We're using `assigned_facility` to group this though. Is that the correct field to use for calculating registrations at a facility?

File: app/queries/my_facilities/blood_pressure_control_query.rb (hunk 3 above)
Comment (prabhanshuguptagit) [reply app/queries/my_facilities/blood_pressure_control_query.rb]: Addressed in 170a095.

File: app/queries/my_facilities/blood_pressure_control_query.rb [hunk 7]
```diff
@@ -8,7 +8,7 @@ class MyFacilities::BloodPressureControlQuery
   include QuarterHelper
   include MonthHelper
 
-  REGISTRATION_BUFFER = 2.months
+  REGISTRATION_BUFFER = 3.months
```
Comment (prabhanshuguptagit) [inline app/queries/my_facilities/blood_pressure_control_query.rb:11]: [Ref](https://simpledotorg.slack.com/archives/CK95QFJA2/p1596107535207500)

File: app/queries/my_facilities/blood_pressure_control_query.rb (hunk 4 above)
Comment (prabhanshuguptagit) [reply app/queries/my_facilities/blood_pressure_control_query.rb]: Right, the `.where(:assigned_facility: facilities)` is incorrect indeed. Addressed in 23d8339.

**Pull request merged**